pkgs/servers/http/nginx/generic.nix

```diff
@@ -145,6 +152,12 @@ stdenv.mkDerivation {
 
   enableParallelBuilding = true;
 
+  nginx_doc = nginx-doc;
+  preInstall = ''
+    mkdir -p $doc
+    cp -r $nginx_doc/* $doc
```

Comment:
```suggestion
  preInstall = ''
    mkdir -p $doc
    ln -s ${nginx_doc}/* $doc
```

Reply:
I don't like it. I interacts poorly with `overrideAttrs`.

Reply:
Fine.

Reply:
You can easily change the doc by overriding nginx-doc. The current code is not that nice.

Reply:
It is a bit trickier due double `callPackage`, but okay, I can do it with overrides.